packages/itwinui-react/src/core/RadioTiles/RadioTile.tsx

```diff
@@ -20,6 +20,30 @@ type RadioTileProps = {
    * Additional description, if needed.
    */
   description?: React.ReactNode;
+  /**
+   *  Assigns disabled styling to component.
+   */
+  disabled?: boolean;
```

Comment:
i don't think `disabled` should be added here.

all input props (`disabled`, `checked`, `defaultChecked`, `onChange`) should now be passed through `inputProps` because `...rest` is not applied on the input

Reply:
but this would be kind of an annoying breaking change.

(this is a result of the other suggestion from https://github.com/iTwin/iTwinUI/pull/1409#discussion_r1267015293.)

now i'm thinking it would be better to go the other way - remove `inputProps` and add `wrapperProps`. it would still be a breaking change, but would only break custom styles (rather than breaking functionality)

pinging @gretanausedaite again for second opinion, as i'm unsure.

@xman343 you can work on a different PR in the meantime.

Reply:
When I remove `disabled` from here, it no longer recognizes the property, even when I pass it through `inputProps`. 
![disabledError1](https://github.com/iTwin/iTwinUI/assets/44583237/4fb11fed-3e73-4650-b476-20ae9541ed70)
![disabledError2](https://github.com/iTwin/iTwinUI/assets/44583237/909eeb0b-ef55-4a8f-8d71-3e45e53a07d0)

Reply:
After deeper investigation of RadioTile my official opinion is:
- We should to move `background-color` and `outline` from `iui-radio-tile-content` to `iui-radio-tile`. This way styling wrapper makes the most sense as currently users can't do much when styling it.
- Keep `ref` and `disabled` on `input` - focus styles are on input and it makes the most sense for user to have this ref.
- Have `style` and `classname` props on wrapper
- Have inputProps for styles and classnames (mainly for other input props) for input but not sure users can do much with input component inside radio-tile.

So RadioTile props looks like this:
``` ts
{
  disabled?: boolean;
  inputProps?: inputStuff;
  <...>
}
```

Component:
``` js
<Box className={cx('iui-radio-tile', className)} style={style} {...rest}>
  <Box as='input' ref={ref} disabled={disabled} {...inputProps} />
  <...>
</Box>
```

Reply:
>     * Keep `ref` and `disabled` on `input` - focus styles are on input and it makes the most sense for user to have this ref.
> 
>     * Have `style` and `classname` props on wrapper

I have to disagree with these conflicting ideas put together. It is incredibly confusing for some things (like `ref` and `disabled`) to go on the input and the others (like `className`/`style`/`onClick`) to go on the wrapper and yet others (like `onChange`/`checked` to go inside `inputProps`).

What's the issue with going with the same approach as Checkbox (#1371)?

Reply:
The current RadioTile setup can only be customized by changing `iui-radio-tile-content` part. 
There is no issue with going with the same approach as Checkbox, just I'm not sure if it's useful at all. Even if we decide to add wrapperProps, I'm still suggesting to move background and outline to the wrapper.

So yes, we can do wrapperProps. Just make sure users can actually customise it.

Reply:
Update: I opened https://github.com/iTwin/iTwinUI/pull/1420 to remove content div and move relevant styles to wrapper div.

@xman343 You might want to change your base branch to `mayank/radio-tile-refactor`.

Reply:
Am I just focusing on that branch now, or am I rebasing this branch onto `mayank/radio-tile-refactor`? How should I do this in VSCode? (I can ask someone else for more detailed instructions if that's preferable.)

And should I remove `inputProps` and add `wrapperProps` before or after I switch base branches?

Reply:
you will still work in your own branch.

1. change the base branch to `mayank/radio-tile-refactor` using github ui
   <img width="575" alt="" src="https://github.com/iTwin/iTwinUI/assets/9084735/c3b8ef8c-4908-4040-9ae8-8b67276b3276">

2. run `git merge origin/mayank/radio-tile-refactor` locally and resolve any merge conflicts, then push.

3. remove `inputProps` after that.

Reply:
Should I accept "incoming" or "current" changes? 
![image](https://github.com/iTwin/iTwinUI/assets/44583237/a3e4b788-4d93-4dde-abd8-cae6910e690e)

Reply:
I'm pretty sure it's incoming, since that's coming from the refactor, but it's kind of hard for me to tell whether that accounts for all the changes we want in the resultant file

Reply:
you need to resolve merge conflicts in a way that makes sense. usually it will be a combination of Incoming and Current. it doesn't have to be perfect as you can always make any corrections in subsequent commits.

Reply:
Ok, completed the merge & removed `inputProps`. Placing `wrapperProps` is next.

Reply:
Added wrapperProps

Reply:
your entire PR is wrong at this point.

currently:
- `className`/`style` are being set on the wrapper, rather than the input.
- none of the class names are being merged. user-passed classes are completely replacing iui classes.

Reply:
Replacing it with the version from `radio-tile-refactor`. I think it'll be easier to fix if I start with that and go from there.

Reply:
Moved `style` to `input`

Reply:
Just to clarify: Do we want `className` and `style` to be on the input, like in `Checkbox`, or do we want them to be on the wrapper, per Greta's suggestion? https://github.com/iTwin/iTwinUI/pull/1409#discussion_r1269139797

Reply:
I read the whole thread. It was just hard to tell from context, and I wanted to be completely sure I was doing the right thing, so that I don't make any further mistakes.

Reply:
`style` & `className` now in input

Reply:
Should I place `iconProps`, `labelProps`, and `subLabelProps` back into `RadioTile`, or should we keep those out? I got the impression from the thread that we're just doing `disabled` and `wrapperProps` now, but I just want to be 100% positive before I update the changeset.

Reply:
> Should I place `iconProps`, `labelProps`, and `subLabelProps` back into `RadioTile`

yes.

also you are still not correctly merging the classes on the wrapper

Reply:
Done

Reply:
classes are still not being merged correctly.

The final resulting class must consist of both the original `iui-` className as well as the user's custom passed class. If that sounds confusing, you can try it out in the playground and see for yourself to get a better idea. The unit test should check for the presence of both classes too.

Reply:
Should be fixed now. All the classes merge properly and the unit test evaluates for both the `iui-` and custom className parts of the className.